File: vscode/src/core/env/envFileNaming.test.ts
import * as assert from 'assert';
import { envDisplayName } from './envFileNaming';

describe('envFileNaming', () => {
  it('null suffix becomes Default', () => {
    assert.strictEqual(envDisplayName(null), 'Default');
  });

  it('empty/blank suffix becomes Default', () => {
    assert.strictEqual(envDisplayName(''), 'Default');
    assert.strictEqual(envDisplayName('   '), 'Default');
  });

  it('single segment is title-cased', () => {
    assert.strictEqual(envDisplayName('dev'), 'Dev');
    assert.strictEqual(envDisplayName('prod'), 'Prod');
    assert.strictEqual(envDisplayName('valet'), 'Valet');
  });

  it('dotted segments are split, title-cased and space joined', () => {
    assert.strictEqual(envDisplayName('prod.staging'), 'Prod Staging');
    assert.strictEqual(envDisplayName('dev.local.mac'), 'Dev Local Mac');
  });

  it('existing capitalisation is preserved on non-leading characters', () => {
    assert.strictEqual(envDisplayName('prodQA'), 'ProdQA');
  });

  it('consecutive dots produce no empty segments', () => {
    assert.strictEqual(envDisplayName('dev..local'), 'Dev Local');
  });
});

File: intellij/src/test/kotlin/dev/nylo/plugin/env/EnvFileNamingTest.kt
package dev.nylo.plugin.env

import org.junit.Assert.assertEquals
import org.junit.Test

class EnvFileNamingTest {

    @Test
    fun `null suffix becomes Default`() {
        assertEquals("Default", EnvFileNaming.displayName(null))
    }

    @Test
    fun `empty suffix becomes Default`() {
        assertEquals("Default", EnvFileNaming.displayName(""))
        assertEquals("Default", EnvFileNaming.displayName("   "))
    }

    @Test
    fun `single segment is title-cased`() {
        assertEquals("Dev", EnvFileNaming.displayName("dev"))
        assertEquals("Prod", EnvFileNaming.displayName("prod"))
        assertEquals("Valet", EnvFileNaming.displayName("valet"))
    }

    @Test
    fun `dotted segments are split and title-cased and space joined`() {
        assertEquals("Prod Staging", EnvFileNaming.displayName("prod.staging"))
        assertEquals("Dev Local Mac", EnvFileNaming.displayName("dev.local.mac"))
    }

    @Test
    fun `existing capitalisation is preserved on non-leading characters`() {
        assertEquals("ProdQA", EnvFileNaming.displayName("prodQA"))
    }

    @Test
    fun `consecutive dots produce no empty segments`() {
        assertEquals("Dev Local", EnvFileNaming.displayName("dev..local"))
    }
}
